### PR TITLE
fix: Update documentation Argocd app CRD health with app of apps

### PR DESCRIPTION
**Pull request by LucasBoisserie**
ref #5146, #5423

I also add argocd app crd health to health documentation to facilitate information search

Checklist:

* [X] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [X] The title of the PR states what changed and the related issues number (used for the release note).
* [X] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [X] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [X] Does this PR require documentation updates?
* [X] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [X] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [X] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [X] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

File: docs/operator-manual/health.md [hunk 1]
```diff
@@ -19,6 +19,38 @@ with at least one value for `hostname` or `IP`.
...
+  resource.customizations: |
+    argoproj.io/Application:
+      health.lua: |
+        hs = {}
+        hs.status = "Progressing"
+        hs.message = ""
+        if obj.status ~= nil then
+          if obj.status.health ~= nil then
+            hs.status = obj.status.health.status
+            hs.message = obj.status.health.message
```
Comment (ymmt2005) [inline docs/operator-manual/health.md:48]: To address #5423, `hs.message` should be checked if it is not nil.
https://github.com/argoproj/argo-cd/issues/5423#issuecomment-845314293

File: docs/operator-manual/upgrading/1.7-1.8.md [hunk 2]
```diff
@@ -27,12 +27,14 @@ data:
     argoproj.io/Application:
       health.lua: |
         hs = {}
-        hs.status = "Healthy"
+        hs.status = "Progressing"
```
Comment (alexmt) [inline docs/operator-manual/upgrading/1.7-1.8.md:30]: 🤦 how could I miss it

**Review by alexmt — APPROVED** [review]
Thanks a lot @LucasBoisserie and @ymmt2005 !

I spent so much time trying to understand why app of apps is not working. Very appreciate your fix.

File: docs/operator-manual/upgrading/1.7-1.8.md (hunk 2 above)
Comment (marwaelayyadi) [reply docs/operator-manual/upgrading/1.7-1.8.md:30]: it still not working with me even after changing the status to Progressing, is there any other configurations I need to update?

**Comment by karlschriek** [comment]
```
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: App1
spec:
  project: default
  ...
```

And then set App2 to depend on App1.

```
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: App2
  annotations:
    argocd.argoproj.io/depends-on: default/App1
spec:
  project: default
  ...
```

I can understand how the current sync-wave approach works only with the app-of-apps approach. Because how would I otherwise prevent App2 from syncing before App1? What if App1 doesn't even exist yet? That is actually the use case that we have. We generate a series of Apps dynamically (and since we don't know what they will be beforehand, we can't use app-of-apps) and in an arbitrary order. But we do want to be able to deploy them in a _specific_ order. We would like to be able to annotate App2 depend on App1 (and have it wait forever in App1 never comes into existence) in a similar way as you would use `depends_on` in Terraform for example.

**Pull request merged**